### PR TITLE
feat: add Lighthouse runner extension

### PR DESCRIPTION
**Pull request by adamraine**
~We shouldn't merge this until LH 10.0 is released, but can still discuss until then.~

### CODE REVIEW

File: test/lighthouse/lighthouse-e2e.test.ts [hunk 1]
```diff
@@ -38,129 +41,158 @@ const FLOW_JSON_REGEX = /window\.__LIGHTHOUSE_FLOW_JSON__ = (.*);<\/script>/;
...
+  const lighthouseExtension = new LighthouseRunnerExtension(browser, page);
+  const runner = await createRunner(flow, lighthouseExtension);
+  const result = await runner.run();
+
+  await page.close();
+  await browser.close();
+
+  assert.ok(result);
+
+  return lighthouseExtension.createFlowResult();
```
Comment (adamraine) [inline test/lighthouse/lighthouse-e2e.test.ts:90]: We need to get some secondary result from the extension and this method made the most sense to me.

Another option is to add an `onFlowResultCreated` callback to the extension constructor or something.

File: package.json [hunk 2]
```diff
@@ -72,11 +72,15 @@
...
   "peerDependencies": {
-    "puppeteer": ">=16.2.0"
+    "puppeteer": ">=16.2.0",
+    "lighthouse": ">=10.0.0"
   },
   "peerDependenciesMeta": {
     "puppeteer": {
       "optional": true
+    },
+    "lighthouse": {
```
Comment (OrKoN) [inline package.json:82]: since it's an optional peer dependency we should avoid any static non-type imports

File: rollup.config.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ export default [
         format: 'cjs',
       },
     ],
+    external: [/^lighthouse/],
```
Comment (OrKoN) [inline rollup.config.ts:17]: probably need to exclude Lighthouse in the next definition too?

Comment (adamraine) [reply rollup.config.ts:17]: I tried that, I don't think it changes anything

**Review by OrKoN — APPROVED** [review]

**Pull request merged**